src/components/GlobeOptimized.jsx

```diff
@@ -395,20 +398,44 @@ export default function GlobeOptimized(props) {
       properties.forEach((property) => {
         switch (property) {
           case 'lat':
+            console.log('Rotation -->');
+            console.log(rotation);
             rotation[1] +=
               params.has(property) && !isNaN(params.get(property))
                 ? Number(params.get(property))
                 : 0;
+            //if initial longitude and latitude not in the range rest it
+            if(rotation[1] <-360 || rotation[1] > 360){
```

Comment:
We need to make sure the number here reflects the value people normally use when presenting the geo location. For example, we use 37 degree N (+37), 122 degree W (-122) to locate San Francisco city. So we need to make sure those values can be parsed into the proper rotation value.